### PR TITLE
`Atlas`: Add Competency Relation Logic 

### PR DESCRIPTION
**Pull request by ufukygmr**
What?

  Adds delete by property functionality to the Weaviate client and comprehensive tests for the
  competency relations suggestion endpoint. Also, adds Competency Rleation logic that returns random relations .

  Why?

  The Weaviate client was missing the ability to delete objects by property value, which is needed for
  data cleanup operations. Additionally, the competency relations suggestion endpoint lacked proper
  input/output type validation tests to ensure API contract compliance.

  How?

  1. Weaviate Client Enhancement: Added delete_by_property method that uses Weaviate's delete_many
  operation with property-based filtering. The method validates inputs, checks collection existence, and
   returns the count of successfully deleted objects.
  2. Test Coverage Expansion:
    - Created comprehensive tests for delete_by_property including success cases, input validation,
  error handling, and edge cases
    - Added input/output type validation tests for suggest_competency_relations endpoint to ensure
  proper CompetencyRelationSuggestionResponse structure
    - Enhanced mock infrastructure to support delete operations with failure simulation

  Affected Issues & Feature Proposal

  Addresses the need for property-based deletion functionality in the Weaviate client and improves test
  coverage for API endpoints.

  Checklist

  General

  - I chose a title conforming to the naming conventions for pull requests.

  AtlasML - Machine Learning

  - I updated the machine learning client functionality.
  - I added comprehensive test coverage for new functionality.

  Testing

  Testing Instructions

  Run
  pytest
  in the root of the directory ( Edutelligence/atlas/AtlasMl ).

  Specific test commands:
  # Test new delete functionality
  poetry run pytest tests/test_weaviate.py -k "delete_by_property" -v

  # Test competency relations endpoint
  poetry run pytest tests/routers/test_competency.py -k "suggest_competency_relations" -v

  # Run all tests to ensure nothing is broken
  poetry run pytest tests/ -v

## Summary by CodeRabbit

* **New Features**
  * Added a competency relations suggestion endpoint for courses.

* **Improvements**
  * Expanded example environment variables for integrations and container runs.
  * Normalized ID formats (now integer-based) and improved request logging.
  * Container updates: new service, healthcheck, exposed port, and runtime adjustments.

* **Bug Fixes**
  * Removed token requirement from some competency endpoints for easier access.

* **Tests**
  * Added tests for relation suggestions and delete-by-property; improved delete mocking.

* **Chores**
  * Added Sentry SDK dependency.

### CODE REVIEW

**Review by ardakaraman0 — DISMISSED** [review]
LGTM

**Review by Yhmidi — DISMISSED** [review]
LGTM

File: atlas/AtlasMl/atlasml/app.py [hunk 1]
```diff
@@ -97,3 +113,7 @@ async def validation_exception_handler(request: Request, exc: RequestValidationE
 
 app.include_router(health_router)
 app.include_router(competency_router)
+
+@app.get("/sentry-debug")
```
Comment (MaximilianAnzinger) [inline atlas/AtlasMl/atlasml/app.py:117]: this endpoint should not be merged

**Review by MaximilianAnzinger — CHANGES_REQUESTED** [review]

**Review by kurunbelemir — APPROVED** [review]
looks good to me

**Review by Yhmidi — APPROVED** [review]
LGTM
